test/unit/url.cpp

```diff
@@ -14,6 +14,55 @@
 #include <algorithm>
 #include <sstream>
 
+#ifdef __has_include
+#if __has_include(<format>)
+    #define BOOST_URL_HAS_STD_FORMAT
+#endif
+#if __has_include(<fmt/format.h>)
+    #define BOOST_URL_HAS_FMT_FORMAT
+#endif
+#endif
+
+#ifdef BOOST_URL_HAS_STD_FORMAT
+#include <format>
+#endif
+
+#ifdef BOOST_URL_HAS_FMT_FORMAT
+#include <fmt/format.h>
+#endif
+
+#ifdef BOOST_URL_HAS_STD_FORMAT
+template <>
+struct std::formatter<boost::urls::url>
```

Comment:
If we are really implementing, probably move it to some kind of formatter header somewhere.